### PR TITLE
csi: Replace leader-election parameters

### PR DESCRIPTION
**Pull request by jsafrane**
CSI controller pods contain many sidecars, which run leader election with the default setting (timeout after 15 seconds). Replace
`${LEADER_ELECTION_LEASE_DURATION}`, `${LEADER_ELECTION_RENEW_DEADLINE}` and `${LEADER_ELECTION_RETRY_PERIOD}` with correct OpenShift's values to make the sidecars resilient to API server disruption during update.

It's up to an operator to put these placeholders to their yaml files to benefit from this replacement.

This is a prerequisite for https://bugzilla.redhat.com/show_bug.cgi?id=1990662 - we need to update leader election in all CSI driver operators.

### CODE REVIEW

File: pkg/operator/csi/csidrivercontrollerservicecontroller/csi_driver_controller_service_controller.go [hunk 1]
```diff
@@ -66,6 +72,12 @@ func NewCSIDriverControllerServiceController(
 	optionalInformers = append(optionalInformers, configInformer.Config().V1().Infrastructures().Informer())
 	var optionalManifestHooks []dc.ManifestHookFunc
 	optionalManifestHooks = append(optionalManifestHooks, WithPlaceholdersHook(configInformer))
+	optionalManifestHooks = append(optionalManifestHooks,
+		WithLeaderElectionReplacerHook(
+			leaderelection.LeaderElectionDefaulting(
```
Comment (sttts) [inline pkg/operator/csi/csidrivercontrollerservicecontroller/csi_driver_controller_service_controller.go:77]: ugly indention, hard to read

Comment (sttts) [reply pkg/operator/csi/csidrivercontrollerservicecontroller/csi_driver_controller_service_controller.go:77]: if you moved it into a local variable `libraryGoLeaderElectionsDefaults` is would be trivial to read. Now I had to check two functions to understand.

Comment (jsafrane) [reply pkg/operator/csi/csidrivercontrollerservicecontroller/csi_driver_controller_service_controller.go]: Reworked to a variable

**Pull request merged**